### DISCUSSION TITLE
Adopt the Sphinx `versionadded` directive

### DISCUSSION THREAD

**Discussion by rcomer** [SciTools/iris#6107] [Ideas]
Prompted by a question on Met Office Viva Engage where the user was unable to use the _use_year_at_season_start_ parameter of [add_season_year](https://scitools-iris.readthedocs.io/en/v3.9.0/generated/api/iris.coord_categorisation.html#iris.coord_categorisation.add_season_year), as they had Iris v3.7.

Sphinx provides a directive to indicate when a function or parameter was added, the [`versionadded` directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionadded).

[Example in Matplotlib](https://github.com/matplotlib/matplotlib/blob/e1f88ba6e36ba12c6376d9eb3878e7f62413022a/lib/matplotlib/contour.py#L1821-L1824) renders like [this](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.contour.html#matplotlib.axes.Axes.contour).

I note that Matplotlib has only been using this for a few versions and has **not** attempted to go back and retrospectively add the directive for older functionality.

**Comment by bjlittle** [comment]
@rcomer I've been using this directive on `geovista` since inception to signpost the version when functionality was first introduced. 

I think it's a valuable piece of documentation metadata for both the user and developer 👍

Note that, I've decided to not bother with the maintenance overhead of using the `versionchanged` directive. TBH I don't think the benefit is worth the effort.

**Comment by trexfeathers** [comment]
@bjlittle raises a good point. To have helped the Met Office user we would have needed to either used `versionchanged`, or provide `versionadded` for every parameter of a function. So far our experience in GeoVista has been `versionadded` only for whole operations.

I have wondered whether this could be automated, but I haven't done any investigation.

**Comment by trexfeathers** [comment]
From @scitools/peloton: In the absence of automation, we would probably need to add this to the pull request checklist (which sometimes gets ignored), and rely on reviewers to pick up on missing directives.
